app/src/service/ocr/document/w2.ts

```diff
@@ -0,0 +1,12 @@
+import { createDocumentMatcher } from "@/service/factory";
+
+export const w2Form = createDocumentMatcher("W2 Form", "w2", {
+  employerIdentificationNumber: /\b(\d{2}-\d{7})/,
+  wagesTipsOthers: /(\d+,\d+\.\d{2}|\d+\.\d{2})\s+/,
+  federalIncomeTaxWithheld: /(\d+,\d+\.\d{2}|\d+\.\d{2})$/,
+  employeeAddress: /Shawn Bailey\s*(.*?)\s*\d{3}-\d{2}-\d{4}/,
+  ssn: /\b\d{3}-\d{2}-\d{4}\b/,
+  bottomLines: /(Form W-2 Statement[\s\S]*?)$/,
+});
+
+export type W2FormPatterns = typeof w2Form.patterns;
```

Comment:
The `createDocumentMatcher` located [here](https://github.com/DSACMS/iv-doc-uploader/pull/11/files#diff-d51131a73161318ac3462120862ba2944e7cdee72c51a66ac122e45bf3346bebR29) is responsible for adding the patterns attribute.

Reply:
I didn't understand what you meant at first, but now I do. That would be pretty cool! I think it'll be easy to implement as well. My thought was that for a specific document - that there wouldn't be multiple patterns for one field. If we decide to support "unknown" documentation (maybe outside the scope of this project) then I think having multiple expressions to identify a single piece of information would be handy.